instrumentation/runtime-telemetry/runtime-telemetry-java8/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/java8/internal/ExperimentalFileDescriptor.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.runtimemetrics.java8.internal;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.metrics.Meter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * Registers measurements that generate experimental metrics about file descriptor. This metrics are
+ * experimental, see <a
+ * href="https://github.com/open-telemetry/semantic-conventions/issues/1275">File Descriptor metrics
+ * semantic conventions</a>
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ */
+public class ExperimentalFileDescriptor {
+  /** Register observers for java runtime experimental file descriptor metrics. */
+  public static List<AutoCloseable> registerObservers(OpenTelemetry openTelemetry) {
+    return registerObservers(
+        openTelemetry,
+        OperatingSystemMethods.openFileDescriptorCount(),
+        OperatingSystemMethods.maxFileDescriptorCount());
+  }
+
+  // Visible for testing
+  static List<AutoCloseable> registerObservers(
+      OpenTelemetry openTelemetry,
+      Supplier<Long> openFileDescriptorCount,
+      Supplier<Long> maxFileDescriptorCount) {
+    Meter meter = JmxRuntimeMetricsUtil.getMeter(openTelemetry);
+    List<AutoCloseable> observables = new ArrayList<>();
+
+    if (openFileDescriptorCount != null) {
+      observables.add(
+          meter
+              .upDownCounterBuilder("process.open_file_descriptor.count")
```

Comment:
Since this name isn't exactly the same as `process.open_file_descriptors` that is used by the host metric receiver we might as well go with a different name. I think it would be best to discuss these names and descriptions at the SIG meeting.

Reply:
Ok, how can i propose this as a topic?

Reply:
based on https://github.com/open-telemetry/semantic-conventions/pull/1618/files

> as a working group we believe it is important that `process` namespace and runtime namespace metrics remain separate, because `process` metrics are meant to represent an **OS-level process as the instrumentation source**, whereas runtime metrics represent **the language runtime as the instrumentation source**.

I'd suggest using `jvm.open_file_descriptor.count`

Reply:
@trask do you also have suggestions for the metric descriptions and unit. Currently the unit is `{count}`.

Reply:
good catch: https://github.com/open-telemetry/semantic-conventions/issues/1662

Reply:
for description, I think we can use

> Number of file descriptors in use by the process.

from https://github.com/open-telemetry/semantic-conventions/blob/373a69574ec9605895102205d561b2476972e20c/model/process/metrics.yaml#L82